Description: Add projectM_guide.txt
 Adding a supplementary projectM Guide, sourced from
 a knowledgable user in the community.
 .
 Added in debian/changelog SNAPSHOT release for bionic PPA
 .
 projectm (3.1.3-1~bionic~ppa1~1.gbp2b2c26) UNRELEASED; urgency=medium
 .
   ** SNAPSHOT build @2b2c26cdf088015683a1f7d08deccb0c8e9efaff **
 .
   * debian/control: Add Breaks & Replaces for bionic package migration to PPA
   * Add text/markdown docs to Debian doc-base
   * Add text/html docs to Debian doc-base.
     Note: Linked images imported via pandoc --extract-media=docs_media
Author: James Cuzella <james.cuzella@lyraphase.com>
Origin: other, https://ubuntuforums.org/showthread.php?t=749793
Last-Update: 2020-05-24

--- /dev/null
+++ projectm-3.1.3/projectM_guide.txt
@@ -0,0 +1,98 @@
+Visualisations are great, and on drugs they are positively excellent. Linux is also awesome, but booting into Windows just to watch visualisations sure is a downer on a good trip. :P Many Linux media players come with preset visualisations, but I find them to be pretty boring and wanted to get my favourite Winamp vis, Milkdrop, to run on my Ubuntu 8.10 box.
+
+Over the last couple of trips I've wanted to get this working, but compiling code is the last thing I would want to do (or could do) on DXM, so I've done it just now, and it was very easy. This is mostly copied verbatim from various sources - it's specifically for Ubuntu 8.10 but the links at the end will help you to get it running on different versions/distros and also under Mac OSX.
+
+ProjectM is a standalone visualiser that can work from many sources including pulseaudio. What this means is that you can use any application that outputs sound to generate the visuals. We are compiling the latest version (currently 2.0) from source; if you just can't get it to work, there are older stable releases (1.2); debs and rpms are available from the projectm homepage.
+
+So here we go:
+
+Download the dependencies:
+sudo apt-get install libglew1.5 libglew1.5-dev ftgl-dev libpulse-dev subversion cmake libvisual-0.4-dev libsdl-dev libqt4-dev build-essential
+
+Create the directory and get the projectm source:
+cd ~
+mkdir projectm
+cd projectm
+svn co https://projectm.svn.sf.net/svnroot/projectm/trunk projectM-Trunk
+
+Then run:
+cd projectM-Trunk/src
+ccmake . (note the ".', you must use this)
+
+Now cmake will load. Press “c” to configure ProjectM. Highlight the CMAKE_BUILD_TYPE field with Tab, press enter and type “Release” there. Press enter again. Then highlight the CMAKE_INSTALL_PREFIX field and change /usr/local to /usr/. Then press “c” again to configure it with those parameters.
+
+For Ubuntu 8.10, you will get some errors and need to:
+cd libprojectM
+gedit Renderer.hpp
+
+Edit lines 28-30 from this:
+#include <FTGL/FTGL.h>
+#include <FTGL/FTGLPixmapFont.h>
+#include <FTGL/FTGLExtrdFont.h>
+
+to this: (easiest to just copy this and replace the lines)
+
+#include <FTGL/ftgl.h>
+/**#include <FTGL/FTGLPixmapFont.h> */
+/**#include <FTGL/FTGLExtrdFont.h> */
+
+Save and close
+
+Now use cmake . (again note the ".")
+cd ..
+cmake .
+make
+
+Finally,
+sudo make install
+
+ProjectM should be in Sound and Video in the Applications menu, or can be run with the command projectM-pulseaudio.
+
+
+Now you can download any of the thousands of presets which are floating around the net. A good tip is to 
+
+cd ~/projectm
+mkdir presets 
+cd /usr/share/projectM/presets && sudo ln -s ~/projectm/presets/ others
+
+So you can paste your presets into the ~/projectm/presets folder and they will appear in the menu in projectm. This saves mucking around with sudo to paste into the /usr directory.
+
+A good set of presets called 'Better Life Through Chemicals' can be found here: http://www.bitcore.org/stuff/BLTC2.1.zip
+
+Just download the file, unrar and copy the folder into ~/projectm/presets
+
+
+You'll want to optimise the settings in projectm, particularly the resolution and texture sizes. You want resolution size to match your monitor, and texture sizes that have the same pixel ratio as your monitor. Eg for a 1280x1024 monitor, the pixel ratio is 1.25 (1280/1024) so you could try texture sizes of 50x40 (50/40=1.25) or something. It's trial and error to get work out what your pc will handle. For reference, my test system is a P4 3ghz, 1gb ram and a 256mb 7600gs gfx card. With a resolution of 1680x1050 I can comfotably run textures of 40x25 and this looks good. With a newer PC you could crank out some fantastic looking visualisations. 
+
+There is a bunch of stuff listed in the Ubuntu forums link at the end of this post about configuring settings.
+
+Basic Projectm commands:
+
+m - brings up a menu
+f - toggles fullscreen on/off
+l - "locks" to a particular preset
+y - toggles shuffle mode
+n - next preset
+p - previous preset
+r - selects random preset
+
+F1 - Help menu
+F2 - Toggles song title on/off (doesn't work in libvisual or pulseaudio as far as I can tell)
+F3 - Toggle preset name on/off
+F4 - Toggel rendering info on/off
+F5 - Shows fps
+
+
+Extra info, sources, etc:
+http://ubuntuforums.org/showthread.php?t=749793
+http://zulu1.wordpress.com/?p=5
+http://forums.winamp.com/showthread.php?s=&threadid=197254
+http://projectm.sourceforge.net
+http://www.milkdrop.co.uk/
+
+
+
+If you can't be bothered with all of this, install xfractint, the classic fractal visualiser. Should be able to get it from standard repos:
+sudo apt-get install xfractint
+
+Hope it all works for you. Enjoy!
